pkg/prowgen/prowgen.go

```diff
@@ -268,6 +275,28 @@ func deleteConfigsIfNeeded(ignoreConfigs []string, paths []string, branch string
 	return nil
 }
 
+func SaveProwConfig(openShiftRelease Repository, repository Repository, config shardprowconfig.ProwConfigWithPointers) error {
+	outPath := filepath.Join(openShiftRelease.Org, openShiftRelease.Repo, "core-services", "prow", "02_config", repository.Org, repository.Repo, "_prowconfig.yaml")
+
+	dir := filepath.Dir(outPath)
+
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+		return err
+	}
+	// Going directly from struct to YAML produces unexpected configs (due to missing YAML tags),
+	// so we produce JSON and then convert it to YAML.
+	out, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+	out, err = gyaml.JSONToYAML(out)
+	if err != nil {
+		return err
+	}
```

Comment:
Do you try using `sigs.k8s.io/yaml`? It may properly marshal the config.

Reply:
Good point. Updated it